src/yamlist/main.rs

```
@@ -1,203 +1,58 @@
-                }
-            }
-            "-o" => {
-                arg_index += 1;
-                if arg_index < len {
-                    outname = String::from(&args[arg_index]);
-                } else {
-                    println!("missing 'OUTNAME' arg for output name");
-                }
-            },
-            _ => {
+type Result<T> = std::result::Result<T, ErrorMessage>;
 
-            }
-        }
-        arg_index += 1;
-    }
+fn main() {
+    let args = Args::from_args();
 
-    if !labelname.is_empty() {
-        if let Err(e) = motion_lib::hash40::load_labels(&labelname) {
+    if let Some(ref label_path) = args.get_label() {
+        if let Err(e) = motion_lib::hash40::load_labels(label_path) {
             println!("Error loading labels: {}", e);
         }
     }
 
-    match mode {
-        Mode::Disasm {file} => {
-            let o = if !outname.is_empty() {
-                &outname
-            } else {
-                "out.yml"
-            };
-
-            match convert_to_yaml(&file, o) {
-                Ok(_) => {}
-                Err(y) => {
-                    let e: &dyn Error = y.borrow();
-                    println!("ERROR: {}", e);
-                }
-            }
+    if let Err(y) = match &args.mode {
+        Mode::Disasm {file, ..} => {
+            convert_to_yaml(&file, &args.get_outfile())
         }
-        Mode::Asm {file} => {
-            let o = if !outname.is_empty() {
-                &outname
-            } else {
-                "out.bin"
-            };
-
-            match convert_to_bin(&file, o) {
-                Ok(_) => {}
-                Err(y) => {
-                    let e: &dyn Error = y.borrow();
-                    println!("ERROR: {}", e);
-                }
-            }
+        Mode::Asm {file, ..} => {
+            convert_to_bin(&file, &args.get_outfile())
         }
         Mode::Patch {..} => {
             unimplemented!()
```

I'm not personally a fan of panicking under normal circumstances, we should probably just return an error to actually print. With backtracing enabled it's pretty ugly looking and isn't needed